src/components/Sidebar/PriceBlock/PriceBlock.tsx

```diff
@@ -1,47 +1,121 @@
-import { useState } from "react";
+import { FC, useState, useEffect, useCallback } from "react";
 import Slider from "react-slider";
 
 import SidebarSectionTitle from "../SidebarSectionTitle";
+import { useAppDispatch } from "@hooks/useAppDispatch";
+import { useAppSelector } from "@hooks/useAppSelector";
+import { setSelectedPriceRange } from "@store/reducers/productSlice";
+import {
+  selectFilteredPriceRange,
+  selectPriceRange,
+} from "@store/selectors/productSelectors";
+import { debounce } from "@helpers/debounce";
+import { PRICE_SLIDER_DELAY } from "@constants/debounceDelays";
 
-import "./Price.scss";
+import "./PriceBlock.scss";
 
-const MIN_VALUE = 1;
-const MAX_VALUE = 100;
+export const PriceBlock: FC = () => {
+  const dispatch = useAppDispatch();
+  const [minPrice, maxPrice] = useAppSelector(selectFilteredPriceRange);
+  const [selectedMinPrice, selectedMaxPrice] = useAppSelector(selectPriceRange);
+  const [values, setValues] = useState<number[]>([0, 0]);
+  const [inputValues, setInputValues] = useState<number[]>([0, 0]);
+  const [error, setError] = useState<boolean | null>(null);
 
-export const PriceBlock = () => {
-  const [values, setValues] = useState([MIN_VALUE, MAX_VALUE]);
+  const debouncedSetSelectedPriceRange = useCallback(
+    debounce((newValues: number[]) => {
+      dispatch(setSelectedPriceRange(newValues));
+    }, PRICE_SLIDER_DELAY),
+    [dispatch]
+  );
+
+  useEffect(() => {
+    dispatch(setSelectedPriceRange([minPrice, maxPrice]));
+  }, [minPrice, maxPrice]);
+
+  useEffect(() => {
+    if (
+      //means reset
+      selectedMinPrice === minPrice &&
+      selectedMaxPrice === maxPrice
+    ) {
+      setInputValues([selectedMinPrice, selectedMaxPrice]);
+      setValues([selectedMinPrice, selectedMaxPrice]);
+    }
+  }, [selectedMinPrice, selectedMaxPrice]);
+
+  const handleInputChange = (index: number, value: string) => {
+    const parsedValue = parseFloat(value);
+    const newInputValues = [...inputValues];
+    newInputValues[index] = isNaN(parsedValue) ? 0 : parsedValue;
+
+    setInputValues(newInputValues);
+    setValues(newInputValues);
+
+    dispatch(setSelectedPriceRange(newInputValues));
+  };
+
+  useEffect(() => {
+    const isInvalid =
+      inputValues[0] > inputValues[1] ||
+      inputValues[0] < minPrice ||
+      inputValues[1] > maxPrice;
+
+    setError(isInvalid);
+  }, [inputValues, minPrice, maxPrice]);
```

Comment:
For better understanding, you can create 2 variables here inputMinPrice/inputMaxPrice and use them in the condition